activity/activity.info

```diff
@@ -1,7 +1,6 @@
 [Activity]
 name = Jumble
-activity_version = 22
-host_version = 1
+activity_version = 23
```

Comment:
Is it possible to add a one-liner summary before the next release?

Reply:
Of course.  Do you have one?  Push to this branch.

Reply:
Oops, missed this message. and I don't have a summary :/

Reply:
http://activities.sugarlabs.org/en-US/sugar/addon/4413 has some text that could be used.

Reply:
> In a jumble of 83 objects, the player has to locate 20. Each deal is different. 

I have never played the activity. I will check its appropriateness and create a PR.

Reply:
I'm sure you'll enjoy it.  :grin:

Reply:
I played it. It was nice. :smile: 
----
by the way, while playing the game, is it obvious for the user, that they can choose between images by using the `O` and `N` buttons? (I had read the release notes on activities.sugarlabs.org; so I knew how to do it. I am not sure if a user will be able to easily understand it(

Reply:
Do they need to understand that in order to play the game?  Sometimes features can be hidden and discoverable.

Reply:
No, it would not had been necessary, but the problem in this case was the jumbled images were, most of the time hidden under bigger images, or probably outside the screen with very few pixel on the screen, (which is not easy to make out). In such cases, the only way to continue the game was to use the `O` and `N`. 

I agree. they could be hidden, discoverable, but it might be annoying if image is not on the screen.

Reply:
What display size?  Is the game ever unplayable because of the hiding?

Reply:
1366x768; The game is unplayable (by unplayable, I mean cannot proceed further) without using any of the keys O or N keys. It happens randomly. Probably we should make the random region quite smaller than the original screen so that at any case nothing goes out of the screen display.

Reply:
Can you please show me a screenshot of such a case? I'm not sure if I properly understand the display issue here

Reply:
Is there a toolbar so you can put buttons with shortcuts for O and N?  I've not checked to see what these keys do.

Reply:
There is a toolbar, we could put the next and previous keys there, but as said earlier, it would not bring out the "easter egg" of the game. 



> Can you please show me a screenshot of such a case? I'm not sure if I properly understand the display issue here

I forgot to take a screenshot when it happens, I will try it again and again.

I mean, the object is not in the identifiable, visible range of the screen. Either
* its hidden under another large object
* A few pixels of the object is visible at the edge of the screen, but its not obvious if its the object.

Reply:
I've tested on 1024x768 resolution.  I did get situation where object requested was not visible.  I don't seem to get any response from right-click, O, or N keys.

Original was for 1200x900 resolution.  So the space has changed.  Perhaps the number of objects should scale according to the reduced space.  Does it?  I've not checked.

Perhaps the objects should never be drawn on the edge of the display.

Perhaps the objects should be drawn in the reverse stack order as the challenge?  That way the object will always be visible.  Seems a bit easy though.

Reply:
Yes I agree. That would do good. 

> Perhaps the objects should never be drawn on the edge of the display.

We could provide a safe display area, subtracting / adding some pixels from `Screen.height()` and `Screen.width()`

> Perhaps the objects should be drawn in the reverse stack order as the challenge? That way the object will always be visible. Seems a bit easy though.

Agreed. I am not sure if this is what you have suggested, but this is what came to my mind: Sort images on the basis of size of the pixmap in descending order, and then add it. Or did you have any other idea?

Yes, that would solve the issue!! :tada:

Reply:
I think the game was initailly designed for 1024x768 resolution. Guessing from [this line](https://github.com/sugarlabs/jumble-activity/blob/35d66c90722583d023cd7d52cf9571725b5d46ba/Jumble.py#L179).

> Perhaps the number of objects should scale according to the reduced space.

The object images are loaded from utils method. They are scaled according to their size. You can see in [this line](https://github.com/sugarlabs/jumble-activity/blob/35d66c90722583d023cd7d52cf9571725b5d46ba/utils.py#L82). So this can't be a possible solution.

Another thing I noticed while testing now was: two cursors are seen. Way to reproduce- move the mouse to the toolbar. The mouse cursor on the toolbar is different and larger from the one that is left behind on the game canvas.

Reply:
> > Perhaps the number of objects should scale according to the reduced space.
> 
> The object images are loaded from utils method. They are scaled according to their size. You can see in [this line](https://github.com/sugarlabs/jumble-activity/blob/35d66c90722583d023cd7d52cf9571725b5d46ba/utils.py#L82). So this can't be a possible solution.

You've misunderstood.  I don't mean scale the object images, I mean scale the number of objects.

> Another thing I noticed while testing now was: two cursors are seen. Way to reproduce- move the mouse to the toolbar. The mouse cursor on the toolbar is different and larger from the one that is left behind on the game canvas.

That's normal for pygame activities.  It's another reason not to use pygame.

Reply:
> That's normal for pygame activities. It's another reason not to use pygame.

Ohhh, thanks! I treated this as an issue in CowBulls [#17](https://github.com/sugarlabs/CowBulls-activity/pull/17) and didn't find a fix, seems like I don't have to treat it as an issue. :smile: